src/java.base/share/classes/sun/net/www/protocol/http/DigestAuthentication.java

```diff
@@ -60,19 +63,50 @@
     private static final String compatPropName = "http.auth.digest." +
         "quoteParameters";
 
+    // default set of disabled message digest algorithms that may not be
+    // used for proxy connections, or plain text http server connections
+
+    // This will probably be expanded to include SHA-1 eventually
+    private static final Set<String> defDisabledAlgs =
+        Set.of("MD5");
```

Comment:
What I'm suggesting is that the content of this set could be seeded with the value of a Security Property, defined in java.security - rather than have the default value hardcoded here.

Reply:
Yes, I think that should work